tests/unit/package_managers/yarn/test_main.py

```diff
@@ -0,0 +1,93 @@
+import re
+from typing import Optional, Union
+from unittest import mock
+
+import pytest
+import semver
+
+from cachi2.core.errors import PackageRejected, UnexpectedFormat
+from cachi2.core.package_managers.yarn.main import _configure_yarn_version
+
+
+@pytest.mark.parametrize(
+    "yarn_path_version, package_manager_version",
+    [
+        pytest.param(semver.VersionInfo(1, 0, 0), None, id="valid-yarnpath-no-packagemanager"),
+        pytest.param(None, semver.VersionInfo(1, 0, 0), id="no-yarnpath-valid-packagemanager"),
+        pytest.param(
+            semver.VersionInfo(1, 0, 0),
+            semver.VersionInfo(1, 0, 0),
+            id="matching-yarnpath-and-packagemanager",
+        ),
+        pytest.param(
+            semver.VersionInfo(1, 0, 0),
+            semver.VersionInfo(
+                1, 0, 0, build="sha224.953c8233f7a92884eee2de69a1b92d1f2ec1655e66d08071ba9a02fa"
+            ),
+            id="matching-yarnpath-and-packagemanager-with-build",
+        ),
+    ],
+)
+@mock.patch("cachi2.core.package_managers.yarn.main.get_semver_from_package_manager")
+@mock.patch("cachi2.core.package_managers.yarn.main.get_semver_from_yarn_path")
+def test_configure_yarn_version(
+    mock_yarn_path_semver: mock.Mock,
+    mock_package_manager_semver: mock.Mock,
+    yarn_path_version: Optional[semver.version.Version],
+    package_manager_version: Optional[semver.version.Version],
+) -> None:
+    mock_project = mock.Mock()
+    mock_yarn_path_semver.return_value = yarn_path_version
+    mock_package_manager_semver.return_value = package_manager_version
+
+    _configure_yarn_version(mock_project)
+
+    if package_manager_version is None:
```

Comment:
Small detail, but should we assert that the `package_manager_version` stays the same and the file was not written in case `package_manager_version is None` resolves to False? Otherwise, the test is only making sure that the function gets executed without errors for 3 of the scenarios.

Reply:
Added something for this :+1: 

I think the right way to do this would be an assertion that the setter for the package_manager property was not called, but I played around with it for a while and couldn't figure out how to make it work :man_facepalming: 

If anyone else knows how to do it, let me know

Reply:
This LGTM